frontend/src/store/controller/selectors.ts

```diff
@@ -201,46 +201,45 @@ export const getAlarmLimitsRequestDraft = createSelector(
   getAlarmLimits,
   (alarmLimits: AlarmLimitsRequestResponse): AlarmLimitsRequest | null => alarmLimits.draft,
 );
-const alarmLimitsCurrentSelector = (key: string) =>
-  createSelector(getAlarmLimitsCurrent, (alarmLimitsCurrent: AlarmLimits | null): Range | null =>
-    alarmLimitsCurrent === null
-      ? null
-      : ((alarmLimitsCurrent as unknown) as Record<string, Range>)[key],
+type OptionalAlarmLimitsLike = AlarmLimits | AlarmLimitsRequest | null;
+type AlarmLimitsSelector = OutputSelector<
+  StoreState,
+  AlarmLimits | null,
+  (res: AlarmLimitsRequestResponse) => OptionalAlarmLimitsLike
+>;
+const rangeSelector = (alarmLimitsSelector: AlarmLimitsSelector) =>
```

Comment:
What I had proposed was that `rangeSelector` would take two arguments, the first is a selector which would be either getAlarmLimitsCurrent, getAlarmLimitsRequest, or getAlarmLimitsRequestDraft, and the second is the key. However, if you no longer need getAlarmLimits*SpO2 and getAlarmLimits*HR, then we don't actually need `rangeSelector`. Right now all it does is cast your `OptionalAlarmLimitsLike` into a `Record<string, Range> | null`, and we don't need a selector to do that.

Reply:
Yep i realised later that we don't really need getAlarmLimitsSpO2 or HR, I have removed the `rangeSelector `now